modules/consensus/accept.go

```diff
@@ -24,8 +25,40 @@ var (
 	errOrphan                 = errors.New("block has no known parent")
 )
 
-// validHeader does some early, low computation verification on the block.
-func (cs *ConsensusSet) validHeader(tx *bolt.Tx, b types.Block) error {
+type (
+	// dbBucket represents a collection of key/value pairs inside the database.
+	dbBucket interface {
+		Get(key []byte) []byte
+	}
+
+	// dbTx represents a read-only transaction on the database that can be used
+	// for retrieving values.
+	dbTx interface {
+		Bucket(name []byte) dbBucket
+	}
+
+	txBlock interface {
+		ID() types.BlockID
+		GetParentID() types.BlockID
+		CheckMinerPayouts(height types.BlockHeight) bool
+		CheckTarget(target types.Target) bool
+		GetTimestamp() types.Timestamp
+	}
+
+	// boltTxWrapper wraps a bolt.Tx so that it matches the dbBucket interface.
+	boltTxWrapper struct {
+		tx *bolt.Tx
+	}
+)
```

Comment:
Can you be more specific? I looked at database.go and I don't see functionality there that does the same thing as I'm doing here (i.e. defining interfaces to wrap `bolt.Tx`).